components/_helpers.js

```diff
@@ -67,8 +67,25 @@ function normalizeDate(dateObject, hours = 0, minutes = 0, seconds = 0, millisec
   return dateObject;
 }
 
+/**
+ * Eject real custom users props from props
+ *
+ * @method ejectOtherProps
+ * @param {Object} props conponent props.
+ * @param {Object} propTypes conponent props types.
+ * @return {Object} custom props.
+ */
+function ejectOtherProps(props, propTypes) {
+  return Object.keys(props)
+    .filter(x => Object.keys(propTypes).indexOf(x) < 0)
```

Comment:
Since you went fancy in the line below, you can go fancy here and do:

```js
const propTypesKeys = Object.keys(propTypes);
Object.keys(props)
  .filter(x => propTypesKeys.includes(x))
  // ...
```

Reply:
@mAiNiNfEcTiOn ``` .filter(x => !propTypesKeys.includes(x)) ``` ;)